### ISSUE TITLE
Bug: TypeError 'Metric object is not subscriptable' in Advanced_Model_Training.ipynb

### ISSUE THREAD

**Issue by GDGouravDey** [deepchem/deepchem#4356]
## 🐛 Bug

The `Advanced_Model_Training.ipynb` notebook in the `./examples/tutorials` directory fails when executing the cell corresponding to the "Early Stopping" section. The error is caused by an incorrect metric indexing in the callback function.

## To Reproduce

1. Open the `Advanced_Model_Training.ipynb` notebook from the `./examples/tutorials` folder in Google Colab.
2. Set the runtime to use a T4 GPU.
3. Run the cell associated with the "Early Stopping" section.
4. An error is raised when the callback attempts to index the metric.

![Image](https://github.com/user-attachments/assets/b844a3b6-8b72-44ed-9da0-38b804185abe)

## Expected behavior

The "Early Stopping" section should execute without errors, with the callback properly accessing the metric by using the correct indexing syntax.

## Environment

* OS: Ubuntu 22.04.4 LTS ( Google Colab )
* Python version: 3.11.11
* DeepChem version: 2.8.1.dev
* RDKit version (optional): 2024.09.6
* TensorFlow version (optional): 2.18.0
* PyTorch version (optional): 2.5.1+cu124
* Any other relevant information: The notebook is executed on Google Colab using a T4 GPU.
